targets/TARGET_Cypress/TARGET_PSOC6/mbed_rtx.h

```diff
@@ -21,6 +21,10 @@
 #ifndef MBED_MBED_RTX_H
 #define MBED_MBED_RTX_H
 
-/* This file is required, but doesn't actually need to do anything. */
+/*
+ * This is used to obtain Object specific Memory allocation required for
+ * mutex creation in an efficient way.
+ */
+#define OS_MUTEX_OBJ_MEM    1
```

Comment:
Is this the best place to add this? This seems like code any TFM_TWINCPU target would need to add, and not PSoC 6 specific. Can we give folks adding new TF-M targets to Mbed OS an easier time of things (i.e. require less copy paste code of them)?

As more acceptable option, we could add an `#ifdef` in `rtos/source/TARGET_CORTEX/mbed_rtx_conf.h` to configure the necessary memory pools (probably eventually more than just mutexes) for all `TFM_TWINCPU` at once, making porting new targets easier.

Reply:
This is more of architectural decision, rather not alter - ```rtos/source/TARGET_CORTEX/mbed_rtx_conf.h``` , but need to find a commonplace that could influence the right platform if the above is not suitable.

Reply:
I don't think adding this configuration for each target is acceptable from a "make adding new hardware easier" perspective. We need to find a shared place to inject RTX configuration from.

Reply:
Just to be clear, In this case, it adds to all Cypress platform.